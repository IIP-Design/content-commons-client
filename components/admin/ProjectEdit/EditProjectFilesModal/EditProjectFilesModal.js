import React, { Fragment, useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import compose from 'lodash.flowright';
import dynamic from 'next/dynamic';
import { graphql } from '@apollo/client/react/hoc';
import {
  Form, Button, Modal, Header, Dimmer,
} from 'semantic-ui-react';

import ButtonAddFiles from 'components/ButtonAddFiles/ButtonAddFiles';
import DynamicConfirm from 'components/admin/DynamicConfirm/DynamicConfirm';
import FileUploadProgressBar from '../../FileUploadProgressBar/FileUploadProgressBar';
import Notification from 'components/Notification/Notification';
import { normalize } from 'lib/graphql/normalizers/video';
import { useFileStateManager } from 'lib/hooks/useFileStateManager';
import { VIDEO_USE_QUERY, IMAGE_USE_QUERY } from 'components/admin/dropdowns/UseDropdown/UseDropdown';
import { FilesContext } from './FilesContext';

import './EditProjectFilesModal.scss';

const EditSupportFilesGrid = dynamic( () => import( /* webpackChunkName: "editSupportFilesGrid" */ './EditSupportFilesGrid/EditSupportFilesGrid' ) );
const EditVideoFilesGrid = dynamic( () => import( /* webpackChunkName: "editVideoFilesGrid" */ './EditVideoFilesGrid/EditVideoFilesGrid' ) );

const EditProjectFilesModal = ( {
  title, type, filesToEdit, extensions,
  save, videoUses: { videoUses },
} ) => {
  const [open, setOpen] = useState( false );
  const [saving, setSaving] = useState( false );
  const [upload, setUpload] = useState( false );
  const [confirm, setConfirm] = useState( {} );
  const [allFieldsSelected, setAllFieldsSelected] = useState( false );
  const [step, setStep] = useState( 1 );

  const {
    state: { files, filesToRemove },
    reset,
    updateFileField,
    addFiles,
    removeFile,
  } = useFileStateManager( normalize );

  const allowedExtensions = extensions.join( ',' );

  const notificationStyles = {
    position: 'absolute',
    top: '15px',
    right: '40px',
  };

  const uploadProgressStyles = {
    margin: '-10px 15px 15px 15px',
  };

  /*
  Check to see if all required dropdowns are completed
  when the the files state changes. All fields do not need
  to be checked as some are pre-populated on initialization or
  not applicable. Save button becomes active when all
  complete
  */
  // check allowed extensions
  const isComplete = () => {
    if ( type === 'support' ) {
      return files.every( file => !!file.language );
    }

    return files.every( file => !!file.language
      && !!file.videoBurnedInStatus
      && !!file.use
      && !!file.quality );
  };

  useEffect( () => {
    setAllFieldsSelected( isComplete() );
  }, [files] );

  const _addFiles = filesToAdd => {
    let defaultUse = '';

    if ( type === 'video' ) {
      defaultUse = videoUses.find( u => u.name === 'Full Video' );
    }

    addFiles( filesToAdd, defaultUse.id );
  };

  const compareFilenames = ( a, b ) => {
    try {
      return a.name.localeCompare( b.name );
    } catch ( err ) {
      console.log( err );
    }
  };

  const toggleStep = () => {
    setStep( s => ( s === 1 ? 2 : 1 ) );
  };

  const closeConfirm = () => {
    setConfirm( { open: false } );
  };

  const checkForDuplicates = ( currentFiles, filesToAdd ) => {
    try {
      const duplicates = filesToAdd.filter( file => currentFiles.includes( file.name ) );

      // if duplicates are present, ask user if they are indeed duplicates
      if ( duplicates.length ) {
        const dups = duplicates.reduce( ( acc, cur ) => `${acc} ${cur.name}\n`, '' );

        setConfirm( {
          open: true,
          headline: 'It appears that duplicate files are being added.',
          content: `Do you want to add these files?\n${dups}`,
          cancelButton: 'No, do not add files',
          confirmButton: 'Yes, add files',
          onCancel: () => {
            _addFiles( filesToAdd.filter( file => !currentFiles.includes( file.name ) ) );
            closeConfirm();
          },
          onConfirm: () => {
            _addFiles( filesToAdd );
            closeConfirm();
          },
        } );
      } else {
        _addFiles( filesToAdd );
      }
    } catch ( err ) {
      console.error( err );
    }
  };

  const closeModal = () => {
    setOpen( false );
  };

  const openModal = () => {
    reset();
    _addFiles( filesToEdit );
    setOpen( true );
    setStep( 1 );
  };

  const updateField = ( e, data ) => {
    updateFileField( data );
  };

  const showFileErrors = uploadedFileErrors => {
    const errors = uploadedFileErrors.reduce( ( acc, cur ) => `${acc} ${cur.name}\n`, '' );
    const multiple = uploadedFileErrors.length > 1;

    setConfirm( {
      open: true,
      headline: `There was an error processing the following file ${multiple ? 's' : ''}`,
      content: errors,
      cancelButton: 'Close',
      confirmButton: 'OK',
      onCancel: () => {
        closeConfirm();
        closeModal();
      },
      onConfirm: () => {
        closeConfirm();
        closeModal();
      },
    } );
  };


  /**
   * Puts file in queue to remove
   * Only show confirm dialogue for files that have already been saved to db
   * For new files, the id is generated using uuid and therefore will
   * have the '-' character in it. Ff '-' exists, assume new file and do not show dialogue
   *
   * @param {string} id id of file
   * @param {string} name name of file
   */
  const handleRemove = ( id, name ) => {
    const matches = id.match( /-/gi );

    if ( !matches ) {
      setConfirm( {
        open: true,
        headline: 'The following file will be removed upon SAVE. Ok?',
        content: name,
        cancelButton: 'No, take me back',
        confirmButton: 'Yes, remove',
        onCancel: () => closeConfirm(),
        onConfirm: () => {
          removeFile( id );
          closeConfirm();
        },
      } );
    } else {
      removeFile( id );
    }
  };

  const handleAddFiles = e => {
    const filesToAdd = Array.from( e.target.files );

    if ( files.length ) {
      const currentFiles = files.map( file => file.name );

      checkForDuplicates( currentFiles, filesToAdd );
    } else {
      _addFiles( filesToAdd );
    }
  };

  const handleSave = async () => {
    setSaving( true );

    const uploadedFiles = files.filter( file => file.input );

    // if there are files to upload, show progress bar
    setUpload( uploadedFiles.length );

    await save( files, filesToRemove );

    setSaving( false );

    const uploadedFileErrors = uploadedFiles.filter( file => file.error );

    if ( uploadedFileErrors.length ) {
      showFileErrors( uploadedFileErrors );
    } else {
      closeModal();
    }
  };

  const renderGrid = () => {
    const sortedFiles = files.sort( compareFilenames );

    if ( type === 'support' ) {
      return (
        <EditSupportFilesGrid
          files={ sortedFiles }
          update={ updateField }
          removeFile={ handleRemove }
          accept={ allowedExtensions }
        />
      );
    }

    return (
      <EditVideoFilesGrid
        files={ sortedFiles }
        update={ updateField }
        removeFile={ handleRemove }
        accept={ allowedExtensions }
        step={ step }
      />
    );
  };

  return (
    <Modal
      open={ open }
      onClose={ closeModal }
      className="edit-support-files"
      size="small"
      closeOnDimmerClick={ !saving }
      trigger={ <Button className="btn--edit" onClick={ openModal } size="small" basic>Edit</Button> }
    >
      <Header content={ title } />

      { saving
        && (
          <Fragment>
            <Notification
              el="p"
              show
              icon
              customStyles={ notificationStyles }
              msg="Saving changes..."
            />

            { !!upload
            && (
              <FileUploadProgressBar
                filesToUpload={ files.filter( file => file.input ) }
                fileProgressMessage
                barSize="small"
                customStyles={ uploadProgressStyles }
              />
            ) }
          </Fragment>
        ) }

      <Dimmer.Dimmable dimmed={ saving }>
        <Dimmer inverted simple />

        <Modal.Content>
          <FilesContext.Provider value={ files }>
            <Form>
              { ' ' }
              { renderGrid() }
              { ' ' }
            </Form>
          </FilesContext.Provider>
        </Modal.Content>
        <Modal.Actions>
          <Button
            className="secondary alternative"
            content="Cancel"
            onClick={ closeModal }
          />

          <ButtonAddFiles
            onChange={ handleAddFiles }
            multiple
            accept={ allowedExtensions }
            className="secondary"
          >
            Add Files
          </ButtonAddFiles>

          { type === 'video' && (
            <Button
              className="secondary"
              type="button"
              content={ step === 1 ? 'Next' : 'Previous' }
              onClick={ toggleStep }
            />
          ) }

          <Button
            className="primary"
            type="button"
            content="Save"
            onClick={ handleSave }
            disabled={ !allFieldsSelected }
          />
        </Modal.Actions>

      </Dimmer.Dimmable>
      <DynamicConfirm { ...confirm } />
    </Modal>
  );
};

EditProjectFilesModal.propTypes = {
  title: PropTypes.string,
  type: PropTypes.string,
  filesToEdit: PropTypes.array,
  extensions: PropTypes.array,
  videoUses: PropTypes.object,
  imageUses: PropTypes.object,
  save: PropTypes.func,
};

export default compose(
  graphql( VIDEO_USE_QUERY, { name: 'videoUses' } ),
  graphql( IMAGE_USE_QUERY, { name: 'imageUses' } ),
)( EditProjectFilesModal );
