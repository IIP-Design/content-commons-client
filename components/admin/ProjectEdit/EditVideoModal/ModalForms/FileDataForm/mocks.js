import * as query from './FileDataFormQueries';
import { LANGUAGES_QUERY, LANGUAGE_BY_NAME_QUERY } from 'components/admin/dropdowns/LanguageDropdown/LanguageDropdown';
import { VIDEO_UNIT_QUERY } from 'components/admin/ProjectEdit/EditVideoModal/ModalSections/FileSection/FileSection';
import { VIDEO_USE_QUERY } from 'components/admin/dropdowns/UseDropdown/UseDropdown';

const props = {
  fileCount: 2,
  setFieldValue: jest.fn(),
  values: {
    language: 'ck2lzfx710hkq07206thus6pt',
    quality: 'WEB',
    use: 'ckcm6aii005dv0720fapgz92p',
    videoBurnedInStatus: 'CLEAN',
    vimeo: 'https://vimeo.com/438865910',
    youtube: '',
  },
};

const reduxState = {
  projectUpdated: false,
};

const mocks = [
  {
    request: {
      query: query.VIDEO_PROJECT_QUERY,
      variables: {
        id: 'ckcoositc0biv0720dwlj4l0g',
      },
    },
    result: {
      data: {
        project: {
          id: 'ckcoositc0biv0720dwlj4l0g',
          units: [
            {
              id: 'ckcoospdp0bjc07207x6j8xhw',
              language: {
                id: 'ck2lzfx710hkq07206thus6pt',
                locale: 'en-us',
                __typename: 'Language',
              },
              __typename: 'VideoUnit',
            },
            {
              id: 'ckcovuyt30co70720tp5w6oem',
              language: {
                id: 'ck2lzfx710hkp07206oo0icbv',
                locale: 'fr-fr',
                __typename: 'Language',
              },
              __typename: 'VideoUnit',
            },
          ],
          __typename: 'VideoProject',
        },
      },
    },
  },
  {
    request: {
      query: query.VIDEO_FILE_QUERY,
      variables: {
        // props.selectedFile from redux
        id: 'ckcoospf10bjh0720ftniacm3',
      },
    },
    result: {
      data: {
        file: {
          id: 'ckcoospf10bjh0720ftniacm3',
          createdAt: '2020-07-16T11:05:08.192Z',
          duration: 14.889875,
          filename: 'Mexico City 1.mp4',
          filesize: 15829673,
          quality: 'WEB',
          videoBurnedInStatus: 'CLEAN',
          dimensions: {
            id: 'ckcoosph00bjm07205nrbnxpu',
            height: 1080,
            width: 1920,
            __typename: 'Dimensions',
          },
          language: {
            id: 'ck2lzfx710hkq07206thus6pt',
            displayName: 'English',
            __typename: 'Language',
          },
          stream: [
            {
              id: 'ckcoospi00bjn0720nt6p5ful',
              site: 'vimeo',
              url: 'https://vimeo.com/438865910',
              __typename: 'VideoStream',
            },
          ],
          use: {
            id: 'ckcm6aii005dv0720fapgz92p',
            name: 'Clean',
            __typename: 'VideoUse',
          },
          __typename: 'VideoFile',
        },
      },
    },
  },
  {
    request: {
      query: query.VIDEO_FILE_LANG_MUTATION,
      variables: {
        data: {
          language: {
            connect: { id: 'ck2lzfx710hkp07206oo0icbv' },
          },
        },
        where: { id: 'ckcoospf10bjh0720ftniacm3' },
      },
    },
    result: {
      data: {
        updateVideoFile: {
          id: 'ckcoospf10bjh0720ftniacm3',
          language: { id: 'ck2lzfx710hkp07206oo0icbv' },
        },
      },
    },
  },
  {
    request: {
      query: query.VIDEO_UNIT_CONNECT_FILE_MUTATION,
      variables: {
        data: {
          files: {
            connect: { id: 'ckcoospf10bjh0720ftniacm3' },
          },
        },
        where: { id: 'ckcoospdp0bjc07207x6j8xhw' },
      },
    },
    result: {
      data: {
        updateVideoUnit: {
          id: 'ckcoospdp0bjc07207x6j8xhw',
          files: { id: 'ckcoospf10bjh0720ftniacm3' },
        },
      },
    },
  },
  {
    request: {
      query: query.VIDEO_UNIT_DISCONNECT_FILE_MUTATION,
      variables: {
        data: {
          files: {
            disconnect: { id: 'ckcoospf10bjh0720ftniacm3' },
          },
        },
        where: { id: 'ckcoospdp0bjc07207x6j8xhw' },
      },
    },
    result: {
      data: {
        updateVideoUnit: {
          id: 'ckcoospdp0bjc07207x6j8xhw',
          files: { id: 'ckcoospf10bjh0720ftniacm3' },
        },
      },
    },
  },
  {
    request: {
      query: query.VIDEO_FILE_SUBTITLES_MUTATION,
      variables: {
        data: {
          videoBurnedInStatus: 'SUBTITLED',
        },
        where: { id: 'ckcoospf10bjh0720ftniacm3' },
      },
    },
    result: {
      data: {
        updateVideoFile: {
          id: 'ckcoospf10bjh0720ftniacm3',
          videoBurnedInStatus: 'SUBTITLED',
        },
      },
    },
  },
  {
    request: {
      query: query.VIDEO_FILE_USE_MUTATION,
      variables: {
        data: {
          use: {
            connect: { id: 'ck2lzfx5y0hhz07207nol9j6r' },
          },
        },
        where: { id: 'ckcoospf10bjh0720ftniacm3' },
      },
    },
    result: {
      data: {
        updateVideoFile: {
          id: 'ckcoospf10bjh0720ftniacm3',
          use: { id: 'ck2lzfx5y0hhz07207nol9j6r' },
        },
      },
    },
  },
  {
    request: {
      query: query.VIDEO_FILE_QUALITY_MUTATION,
      variables: {
        data: {
          quality: 'WEB',
        },
        where: { id: 'ckcoospf10bjh0720ftniacm3' },
      },
    },
    result: {
      data: {
        updateVideoFile: {
          id: 'ckcoospf10bjh0720ftniacm3',
          quality: 'WEB',
        },
      },
    },
  },
  {
    request: {
      query: query.VIDEO_FILE_CREATE_STREAM_MUTATION,
      variables: {
        data: {
          stream: {
            create: {
              site: 'thesite',
              url: 'https://thesite.com/the-site-stream-id',
            },
          },
        },
        where: { id: 'ckcoospf10bjh0720ftniacm3' },
      },
    },
    result: {
      data: {
        updateVideoFile: {
          id: 'ckcoospf10bjh0720ftniacm3',
          stream: { id: 'the-site-stream-id' },
        },
      },
    },
  },
  {
    request: {
      query: query.VIDEO_FILE_UPDATE_STREAM_MUTATION,
      variables: {
        data: {
          stream: {
            update: {
              data: {
                url: 'https://thesite.com/the-new-site-stream-id',
              },
              where: {
                id: 'the-new-site-stream-id',
              },
            },
          },
        },
        where: { id: 'ckcoospf10bjh0720ftniacm3' },
      },
    },
    result: {
      data: {
        updateVideoFile: {
          id: 'ckcoospf10bjh0720ftniacm3',
          stream: { id: 'the-new-site-stream-id' },
        },
      },
    },
  },
  {
    request: {
      query: query.VIDEO_FILE_DELETE_STREAM_MUTATION,
      variables: {
        data: {
          stream: {
            'delete': {
              id: 'the-site-stream-id',
            },
          },
        },
        where: { id: 'ckcoospf10bjh0720ftniacm3' },
      },
    },
    result: {
      data: {
        updateVideoFile: {
          id: 'ckcoospf10bjh0720ftniacm3',
          stream: { id: 'the-site-stream-id' },
        },
      },
    },
  },
  {
    request: {
      query: query.VIDEO_FILE_DELETE_MUTATION,
      variables: {
        id: 'ckcoospf10bjh0720ftniacm3',
      },
    },
    result: {
      data: {
        deleteVideoFile: {
          id: 'ckcoospf10bjh0720ftniacm3',
        },
      },
    },
  },
  {
    request: {
      query: LANGUAGE_BY_NAME_QUERY,
      variables: {
        where: { displayName: 'English' },
      },
    },
    result: {
      data: {
        languages: [
          {
            id: 'ck2lzfx710hkq07206thus6pt',
            displayName: 'English',
            locale: 'en-us',
          },
        ],
      },
    },
  },
  {
    request: {
      query: VIDEO_USE_QUERY,
      variables: {
        where: { name: 'Clean' },
      },
    },
    result: {
      data: {
        videoUses: [
          {
            id: 'ckcm6aii005dv0720fapgz92p',
            name: 'Clean',
          },
        ],
      },
    },
  },
  {
    request: {
      query: VIDEO_UNIT_QUERY,
      variables: { id: 'ckcoospdp0bjc07207x6j8xhw' },
    },
    result: {
      data: {
        videoUnit: {
          id: 'ckcoospdp0bjc07207x6j8xhw',
          files: [
            {
              id: 'ckcoospf10bjh0720ftniacm3',
            },
          ],
          language: {
            id: 'ck2lzfx710hkq07206thus6pt',
            displayName: 'English',
            locale: 'en-us',
          },
        },
      },
    },
  },
  {
    request: {
      query: LANGUAGES_QUERY,
    },
    result: {
      data: {
        languages: [
          {
            id: 'ck2lzfx710hkp07206oo0icbv',
            displayName: 'French',
            locale: 'fr-fr',
          },
          {
            id: 'ck2lzfx710hkq07206thus6pt',
            displayName: 'English',
            locale: 'en-us',
          },
        ],
      },
    },
  },
  {
    request: {
      query: VIDEO_USE_QUERY,
    },
    result: {
      data: {
        videoUses: [
          {
            id: 'ckcm6aii005dv0720fapgz92p',
            name: 'Clean',
          },
          {
            id: 'ck2lzfx5y0hhz07207nol9j6r',
            name: 'Full Video',
          },
        ],
      },
    },
  },
];

export { mocks, props, reduxState };
