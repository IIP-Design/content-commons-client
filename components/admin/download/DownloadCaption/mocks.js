import { VIDEO_PROJECT_PREVIEW_CAPTIONS_QUERY } from './DownloadCaption';

export const props = {
  id: '123',
  instructions: 'Download SRT(s)',
  isPreview: false,
};

export const mocks = [
  {
    request: {
      query: VIDEO_PROJECT_PREVIEW_CAPTIONS_QUERY,
      variables: { id: props.id },
    },
    result: {
      data: {
        project: {
          id: props.id,
          files: [
            {
              id: 'srt89',
              url: `2019/06/${props.id}/srt-1.srt`,
              language: {
                id: 'en23',
                displayName: 'English',
              },
            },
            {
              id: 'srt99',
              url: `2019/06/${props.id}/srt-2.srt`,
              language: {
                id: 'en23',
                displayName: 'English',
              },
            },
          ],
        },
      },
    },
  },
];

export const errorMocks = [
  {
    ...mocks[0],
    result: {
      errors: [{ message: 'There was an error.' }],
    },
  },
];

export const nullProjectMocks = [
  {
    ...mocks[0],
    result: { data: { project: null } },
  },
];

export const emptyProjectMocks = [
  {
    ...mocks[0],
    result: { data: { project: {} } },
  },
];

export const noFilesMocks = [
  {
    ...mocks[0],
    result: {
      data: {
        project: {
          ...mocks[0].result.data.project,
          files: [],
        },
      },
    },
  },
];
