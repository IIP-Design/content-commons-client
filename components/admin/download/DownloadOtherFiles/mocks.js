import { VIDEO_PROJECT_PREVIEW_OTHER_FILES_QUERY } from './DownloadOtherFiles';

export const props = {
  id: '123',
  instructions: 'Download Other File(s)',
  isPreview: false,
};

export const mocks = [
  {
    request: {
      query: VIDEO_PROJECT_PREVIEW_OTHER_FILES_QUERY,
      variables: { id: props.id },
    },
    result: {
      data: {
        project: {
          id: props.id,
          files: [
            {
              id: 'pdf89',
              filename: 'file-1.pdf',
              filetype: 'application/pdf',
              url: `2019/06/${props.id}/file-1.pdf`,
              language: {
                id: 'en23',
                displayName: 'English',
              },
            },
            {
              id: 'au56',
              filename: 'audio-1.mp3',
              filetype: 'audio/x-mpeg-3',
              url: `2019/06/${props.id}/audio-1.mp3`,
              language: {
                id: 'en23',
                displayName: 'English',
              },
            },
          ],
          thumbnails: [
            {
              id: 'th765',
              filename: 'thumbnail-1.jpg',
              filetype: 'image/jpeg',
              url: `2019/06/${props.id}/thumbnail-1.jpg`,
              language: {
                id: 'en23',
                displayName: 'English',
              },
            },
            {
              id: 'th865',
              filename: 'thumbnail-2.jpg',
              filetype: 'image/jpeg',
              url: `2019/06/${props.id}/thumbnail-2.jpg`,
              language: {
                id: 'en23',
                displayName: 'English',
              },
            },
          ],
        },
      },
    },
  },
];

export const errorMocks = [
  {
    ...mocks[0],
    result: {
      errors: [{ message: 'There was an error.' }],
    },
  },
];

export const nullProjectMocks = [
  {
    ...mocks[0],
    result: { data: { project: null } },
  },
];

export const emptyProjectMocks = [
  {
    ...mocks[0],
    result: { data: { project: {} } },
  },
];

export const noFilesMocks = [
  {
    ...mocks[0],
    result: {
      data: {
        project: {
          ...mocks[0].result.data.project,
          files: [],
          thumbnails: [],
        },
      },
    },
  },
];
