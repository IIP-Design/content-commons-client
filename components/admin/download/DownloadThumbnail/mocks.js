import { VIDEO_PROJECT_PREVIEW_THUMBNAILS_QUERY } from './DownloadThumbnail';

export const props = {
  id: '123',
  instructions: 'Download Thumbnail(s)',
  isPreview: false,
};

export const mocks = [
  {
    request: {
      query: VIDEO_PROJECT_PREVIEW_THUMBNAILS_QUERY,
      variables: { id: props.id },
    },
    result: {
      data: {
        project: {
          id: props.id,
          thumbnails: [
            {
              id: 'th765',
              url: `2019/06/${props.id}/thumbnail-1.jpg`,
              language: {
                id: 'en23',
                displayName: 'English',
              },
            },
            {
              id: 'th865',
              url: `2019/06/${props.id}/thumbnail-2.jpg`,
              language: {
                id: 'en23',
                displayName: 'English',
              },
            },
          ],
        },
      },
    },
  },
];

export const errorMocks = [
  {
    ...mocks[0],
    result: {
      errors: [{ message: 'There was an error.' }],
    },
  },
];

export const nullProjectMocks = [
  {
    ...mocks[0],
    result: { data: { project: null } },
  },
];

export const noFilesMocks = [
  {
    ...mocks[0],
    result: {
      data: {
        project: {
          ...mocks[0].result.data.project,
          thumbnails: [],
        },
      },
    },
  },
];

export const emptyProjectMocks = [
  {
    ...mocks[0],
    result: { data: { project: {} } },
  },
];
