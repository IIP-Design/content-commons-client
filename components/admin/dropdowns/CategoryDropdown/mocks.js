export const categories = [
  {
    id: 'cjubblzw815qb0756cfwa9vd7',
    translations: [
      {
        id: 'cjubbldkt0ur20756zjfw5209',
        name: 'about america',
      },
    ],
  },
  {
    id: 'cjubblzw815qc07563nzatqyn',
    translations: [
      {
        id: 'cjubbldsp0uta0756p4as8gl9',
        name: 'arts & culture',
      },
    ],
  },
  {
    id: 'cjubblzw915qd0756808n3h3q',
    translations: [
      {
        id: 'cjubbldz10uvi0756o7pg0lqm',
        name: 'democracy & civil society',
      },
    ],
  },
  {
    id: 'cjubblzw915qe07560zkz9xas',
    translations: [
      {
        id: 'cjubble530uxq0756q5bphm9c',
        name: 'economic opportunity',
      },
    ],
  },
  {
    id: 'cjubblzwk15qk0756nenwp3rz',
    translations: [
      {
        id: 'cjubbleb70uzy0756tay96s62',
        name: 'education',
      },
    ],
  },
  {
    id: 'cjubblzwq15r10756m11zntim',
    translations: [
      {
        id: 'cjubbleh10v2607568m3kisep',
        name: 'environment',
      },
    ],
  },
  {
    id: 'cjubblzwq15r20756m3108d8s',
    translations: [
      {
        id: 'cjubbleli0v4e075641kjd0jj',
        name: 'geography',
      },
    ],
  },
  {
    id: 'cjubblzwq15r30756l8e8zdvu',
    translations: [
      {
        id: 'cjubbler60v6m0756231jvsgj',
        name: 'global issues',
      },
    ],
  },
  {
    id: 'cjubblzwq15r40756kb7neely',
    translations: [
      {
        id: 'cjubblevs0v8u075668mzno2t',
        name: 'health',
      },
    ],
  },
  {
    id: 'cjubblzwr15r507566hzkgmpk',
    translations: [
      {
        id: 'cjubblf0v0vb207568iibzp1t',
        name: 'human rights',
      },
    ],
  },
  {
    id: 'cjubblzwr15r607568f7sy9pz',
    translations: [
      {
        id: 'cjubblf5m0vda07565wesiogo',
        name: 'press & journalism',
      },
    ],
  },
  {
    id: 'cjubblzwr15r70756o5jr69p9',
    translations: [
      {
        id: 'cjubblfap0vfi0756pva51gji',
        name: 'good governance',
      },
    ],
  },
  {
    id: 'cjubblzws15rb0756pi1gv0mn',
    translations: [
      {
        id: 'cjubblffg0vhq0756gutom6cs',
        name: 'religion & values',
      },
    ],
  },
  {
    id: 'cjubblzws15rc0756ypyls4bf',
    translations: [
      {
        id: 'cjubblfkk0vjy0756zr2qwjii',
        name: 'science & technology',
      },
    ],
  },
  {
    id: 'cjubblzx615sc0756oh8pk89e',
    translations: [
      {
        id: 'cjubblfp90vm60756cdzur3vo',
        name: 'sports',
      },
    ],
  },
];
