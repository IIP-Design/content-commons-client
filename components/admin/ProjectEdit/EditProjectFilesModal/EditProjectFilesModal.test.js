import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { data } from '../../ProjectSupportFiles/SupportFileTypeList/mocks';
import { config } from '../../ProjectSupportFiles/VideoProjectSupportFiles/config';
import EditProjectFilesModal from './EditProjectFilesModal';

jest.mock( 'next/dynamic', () => () => 'dynamically-imported-component' );

const props = {
  supportFiles: data.projectFiles.supportFiles,
  extensions: config.supportFiles.types.captions.extensions,
  save: jest.fn(),
};

const Component = <EditProjectFilesModal { ...props } />;

describe( '<EditSupportFiles />', () => {
  it( 'renders without crashing', () => {
    const wrapper = shallow( Component );

    expect( wrapper.exists() ).toEqual( true );
    expect( toJSON( wrapper ) ).toMatchSnapshot();
  } );

  it.skip( 'clicking Save Button calls props.save', () => {
    const wrapper = shallow( Component );
    const saveBtn = wrapper.find( 'Button.primary' );

    saveBtn.simulate( 'click' );
    expect( props.save ).toHaveBeenCalled();
  } );
} );
