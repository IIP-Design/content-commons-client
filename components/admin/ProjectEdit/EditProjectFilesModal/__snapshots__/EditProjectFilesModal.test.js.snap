// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditSupportFiles /> renders without crashing 1`] = `
<Query
  displayName="Apollo(Apollo(EditProjectFilesModal))"
  query={
    Object {
      "definitions": Array [
        Object {
          "directives": Array [],
          "kind": "OperationDefinition",
          "name": Object {
            "kind": "Name",
            "value": "VIDEO_USE_QUERY",
          },
          "operation": "query",
          "selectionSet": Object {
            "kind": "SelectionSet",
            "selections": Array [
              Object {
                "alias": undefined,
                "arguments": Array [],
                "directives": Array [],
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "videoUses",
                },
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "id",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "name",
                      },
                      "selectionSet": undefined,
                    },
                  ],
                },
              },
            ],
          },
          "variableDefinitions": Array [],
        },
      ],
      "kind": "Document",
      "loc": Object {
        "end": 73,
        "start": 0,
      },
    }
  }
  skip={false}
>
  <Component />
</Query>
`;
